src/test/java/com/limechain/storage/crypto/KeyStoreTest.java

```diff
@@ -0,0 +1,36 @@
+package com.limechain.storage.crypto;
+
+import com.limechain.chain.Chain;
+import com.limechain.storage.DBInitializer;
+import com.limechain.storage.DBRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class KeyStoreTest {
+
+    private DBRepository dbRepository;
+
+    private KeyStore keyStore;
+
+    @BeforeEach
+    public void setup() {
+        dbRepository = DBInitializer.initialize("./test", Chain.WESTEND, false);
+        keyStore = new KeyStore(dbRepository);
+    }
+
+    byte[] key = {1,2,3};
+    byte[] value = {1,2,3, 4};
+    @Test
+    void saveAndGetKey() {
```

Comment:
Unit tests should test one specific functionality.

Reply:
I can't really test getting a key, without saving it

Reply:
You can test them separately, if you mock the repository.
By using a real repository, this is more of an integration test, than a unit test.